presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -236,7 +237,9 @@ public final void verifyComparableOrderableContract()
                     missingOperators.put(type, NOT_EQUAL);
                 }
             }
-            if (type.isOrderable()) {
+            if (type.isOrderable() && !(type instanceof EnumType)) {
```

Comment:
I don't like special cases. It doesn't feel necessary. For database people, this behavior would be very unexpected. Either support orderable, or not. I think both are reasonable. The result of enum'A' > enum'B' might not make sense but that's ok I think.

Reply:
Actually `comparable` and `orderable` are two separate concept. If we only want them to be comparable we can do that.

Reply:
I implemented full comparison eventually